### DISCUSSION TITLE
new bridges for different platforms/services

### DISCUSSION THREAD

**Discussion by ghost** [git-bug/git-bug#871] [Q&A] [answered]
Hi all! I would be happy if there are more bridges in git-bug like: [Codeberg](https://codeberg.org/), [Gnu Savannah](https://savannah.gnu.org/), [fedora pagure](pagure.io) or [Radicle](https://radicle.community). But what is needed to create a bridge? Could this be done technically?

**Comment by MichaelMure** [comment, marked as answer]
I welcome bridge contribution, but it's not my personal priority. Implementing a bridge is roughly:
- importer: query the remote tracker for updates, make sure they are replicated in git-bug (identities, bugs ...)
- exporter: find in git-bug what has not been marked as exported, make the API calls to replicate that (which means the remote tracker need to have a reasonable API to do so)

The most advanced and complete bridge currently is [the github one](https://github.com/MichaelMure/git-bug/tree/master/bridge/github), about 2200 lines of code, not counting tests. The gitlab one is 1200 LOC.  It's something doable for someone a bit motivated.

**Reply by ghost** [reply]
Hi MichaelMure, thank you for feedback.
